src/codegen.cpp

```diff
@@ -4017,77 +4005,79 @@ static jl_cgval_t emit_expr(jl_value_t *expr, jl_codectx_t *ctx)
+        Value *a1 = maybe_decay_untracked(boxed(ctx, emit_expr(ctx, args[1])));
+        Value *a2 = maybe_decay_untracked(boxed(ctx, emit_expr(ctx, args[2])));
         Value *mdargs[4] = {
             /*argdata*/a1,
             /*code*/a2,
-            /*module*/literal_pointer_val((jl_value_t*)ctx->module),
-            /*isstaged*/literal_pointer_val(args[3])
+            /*module*/literal_pointer_val(ctx, (jl_value_t*)ctx.module),
+            /*isstaged*/literal_pointer_val(ctx, args[3])
         };
-        builder.CreateCall(prepare_call(jlmethod_func), makeArrayRef(mdargs));
+        ctx.builder.CreateCall(prepare_call(jlmethod_func), makeArrayRef(mdargs));
         return ghostValue(jl_void_type);
     }
     else if (head == const_sym) {
         jl_sym_t *sym = (jl_sym_t*)args[0];
-        jl_module_t *mod = ctx->module;
+        jl_module_t *mod = ctx.module;
         if (jl_is_globalref(sym)) {
             mod = jl_globalref_mod(sym);
             sym = jl_globalref_name(sym);
         }
         if (jl_is_symbol(sym)) {
             JL_FEAT_REQUIRE(ctx, runtime);
             jl_binding_t *bnd = NULL;
-            (void)global_binding_pointer(mod, sym, &bnd, true, ctx); assert(bnd);
-            builder.CreateCall(prepare_call(jldeclareconst_func),
-                               literal_pointer_val(bnd));
+            (void)global_binding_pointer(ctx, mod, sym, &bnd, true); assert(bnd);
+            ctx.builder.CreateCall(prepare_call(jldeclareconst_func),
+                               literal_pointer_val(ctx, bnd));
         }
     }
     else if (head == new_sym) {
-        jl_value_t *ty = expr_type(args[0], ctx);
+        jl_value_t *ty = expr_type(ctx, args[0]);
         size_t nargs = jl_array_len(ex->args);
         if (jl_is_type_type(ty) &&
             jl_is_datatype(jl_tparam0(ty)) &&
             jl_is_leaf_type(jl_tparam0(ty))) {
-            assert(nargs <= jl_datatype_nfields(jl_tparam0(ty))+1);
-            return emit_new_struct(jl_tparam0(ty),nargs,args,ctx);
+            assert(nargs <= jl_datatype_nfields(jl_tparam0(ty)) + 1);
+            return emit_new_struct(ctx, jl_tparam0(ty), nargs, args);
         }
-        Value *typ = boxed(emit_expr(args[0], ctx), ctx);
-        Value *val = emit_jlcall(jlnew_func, typ, &args[1], nargs-1, ctx);
-        return mark_julia_type(val, true, ty, ctx);
+        Value *typ = boxed(ctx, emit_expr(ctx, args[0]));
+        Value *val = emit_jlcall(ctx, jlnew_func, typ, &args[1], nargs - 1);
+        return mark_julia_type(ctx, val, true, ty);
     }
     else if (head == exc_sym) { // *ptls->exception_in_transit
-        return mark_julia_type(builder.CreateLoad(emit_exc_in_transit(ctx),
-                                                  /*isvolatile*/true),
-                               true, jl_any_type, ctx);
+        return mark_julia_type(ctx,
+                ctx.builder.CreateLoad(emit_exc_in_transit(ctx), /*isvolatile*/true),
```

Comment:
seriously, #15591 would be much more efficient at this. It's not worth your time to point these out or my time to fix them, when the computer can do both instantly.